### PR TITLE
[Converter] Add custom python converter testcases

### PR DESCRIPTION
**Pull request by gichan-jang**
Add custom python converter testcases.
 1. Test dynamic dimension
 2. Test custom python conveter using json

Signed-off-by: Gichan Jang <gichan2.jang@samsung.com>

Self evaluation:
1. Build test: [ *]Passed [ ]Failed [ ]Skipped
2. Run test: [ *]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: tests/nnstreamer_converter/unittest_converter.cc [hunk 1]
```diff
@@ -331,6 +333,232 @@ TEST (tensorConverter, subpluginFindInvalidParam_n)
...
+  g_free (str_pipeline);
+  ASSERT_NE (pipeline, nullptr);
+
+  appsrc_handle = gst_bin_get_by_name (GST_BIN (pipeline), "srcx");
+  EXPECT_NE (appsrc_handle, nullptr);
+
+  sink_handle = gst_bin_get_by_name (GST_BIN (pipeline), "sinkx");
+  EXPECT_NE (sink_handle, nullptr);
+
+  g_signal_connect (sink_handle, "new-data", (GCallback)new_data_cb, NULL);
```
Comment (jaeyun-jung) [inline tests/nnstreamer_converter/unittest_converter.cc:395]: suggestion : alloc and pass data_received to new-data cb 
```
data_received = (guint *) g_malloc0 (sizeof (guint));
g_signal_connect (sink_handle, "new-data", (GCallback)new_data_cb, data_received);
```

File: ext/nnstreamer/tensor_converter/tensor_converter_python3.cc [hunk 2]
```diff
@@ -365,18 +365,24 @@ extern "C" {
...
 void
 fini_converter_py (void)
 {
-  /** Python should be initialized and finalized only once */
-  Py_Finalize ();
   unregisterExternalConverter (Python.name);
+  /** Python should be initialized and finalized only once */
+  if (Py_IsInitialized()) {
+    /** Wait external converter is unregistered */
+    g_usleep (100000);
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_converter/tensor_converter_python3.cc:383]: Q. Really necessary? `unregisterExternalConverter` is asynchronous method?

Comment (gichan-jang) [reply ext/nnstreamer/tensor_converter/tensor_converter_python3.cc:383]: The python C-API seems operating asynchronously.
Without sleep time, a segmentation fault occurs after the test passes successfully.
This looks like an old problem with numpy and python C-API.

Comment (dongju-chae) [reply ext/nnstreamer/tensor_converter/tensor_converter_python3.cc:383]: Then, please comments, marking `TODO`

**Review by dongju-chae — APPROVED** [review]
LGTM

File: ext/nnstreamer/tensor_converter/tensor_converter_python3.cc (hunk 2 above)
Comment (gichan-jang) [reply ext/nnstreamer/tensor_converter/tensor_converter_python3.cc:387]: Updated.

**Review by jaeyun-jung — APPROVED** [review]

**Review by myungjoo — COMMENTED** [review]
You have two new positive cases and one new negative case. You need two more negative cases.

**Comment by gichan-jang** [comment]
> You have two new positive cases and one new negative case. You need two more negative cases.

Thanks. One positive TC and five negative TCs are added.

**Review by myungjoo — APPROVED** [review]

**Pull request merged**